x-pack/plugins/stack_connectors/server/connector_types/gemini/gemini.ts

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { ServiceParams, SubActionConnector } from '@kbn/actions-plugin/server';
+import { AxiosError, Method } from 'axios';
+import { SubActionRequestParams } from '@kbn/actions-plugin/server/sub_action_framework/types';
+import { initDashboard } from '../lib/gen_ai/create_gen_ai_dashboard';
+import {
+  RunActionParamsSchema,
+  InvokeAIActionParamsSchema,
+  RunApiLatestResponseSchema,
+} from '../../../common/gemini/schema';
+import {
+  Config,
+  Secrets,
+  RunActionParams,
+  RunActionResponse,
+  InvokeAIActionParams,
+  InvokeAIActionResponse,
+  RunApiLatestResponse,
+} from '../../../common/gemini/types';
+import { SUB_ACTION, DEFAULT_TOKEN_LIMIT } from '../../../common/gemini/constants';
+import {
+  DashboardActionParams,
+  DashboardActionResponse,
+} from '../../../common/gemini/types';
+import { DashboardActionParamsSchema } from '../../../common/gemini/schema';
+
+/** Interfaces to define the Gemini model response type */
+
+interface MessagePart {
+  text: string;
+}
+
+interface MessageContent {
+  role: string;
+  parts: MessagePart[];
+}
+
+interface Payload {
+  contents: MessageContent[];
+  generation_config: {
+      temperature: number;
+      maxOutputTokens: number;
+  };
+}
+
+export class GeminiConnector extends SubActionConnector<Config, Secrets> {
+  private url;
+  private model;
+  private gcpRegion;
+  private gcpProjectID;
+
+  constructor(params: ServiceParams<Config, Secrets>) {
+    super(params);
+
+    this.url = this.config.apiUrl;
+    this.model = this.config.defaultModel;
+    this.gcpRegion = this.config.gcpRegion;
+    this.gcpProjectID = this.config.gcpProjectID;
+
+    this.registerSubActions();
+  }
+
+  private registerSubActions() {
+    this.registerSubAction({
+      name: SUB_ACTION.RUN,
+      method: 'runApi',
+      schema: RunActionParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.DASHBOARD,
+      method: 'getDashboard',
+      schema: DashboardActionParamsSchema,
+    });
+
+    this.registerSubAction({
+      name: SUB_ACTION.TEST,
+      method: 'runTestApi',
```

Comment:
this method should be `runApi`. RUN and TEST should use the exact some method. Any formatting you need to do for the assistant would be done in `invokeAI`